mne/source_tfr.py

```diff
@@ -0,0 +1,367 @@
+        Vertex numbers corresponding to the data.
+    tmin : float
+        Time point of the first sample in data.
+    tstep : float
+        Time step between successive samples in data.
+    freqs : ndarray, shape (n_freqs,)
+        The frequencies in Hz.
+    dims : tuple, default ("dipoles", "freqs", "times")
+        The dimension names of the data, where each element of the tuple
+        corresponds to one axis of the data field. Allowed values are:
+        ("dipoles", "freqs", "times"), ("dipoles", "epochs", "freqs",
+        "times"), ("dipoles", "orientations", "freqs", "times"), ("dipoles",
+         "orientations", "epochs", "freqs", "times").
+    method : str | None, default None
+        Comment on the method used to compute the data, as a combination of
+        the used method and the compued product (e.g. "morlet-power" or
+        "stockwell-itc").
+    subject : str | None
+        The subject name. While not necessary, it is safer to set the
+        subject parameter to avoid analysis errors.
+    %(verbose)s
+
+    Attributes
+    ----------
+    freqs : ndarray, shape (n_freqs,)
+        The frequencies in Hz.
+    method : str | None, default None
+        Comment on the method used to compute the data, as a combination of
+        the used method and the compued product (e.g. "morlet-power" or
+        "stockwell-itc").
+    subject : str | None
+        The subject name.
+    times : array, shape (n_times,)
+        The time vector.
+    vertices : array | list of array of shape (n_dipoles,)
+        The indices of the dipoles in the different source spaces. Can
+        be an array if there is only one source space (e.g., for volumes).
+    data : array of shape (n_dipoles, n_times)
+        The data in source space.
+    dims : tuple
+        The dimension names corresponding to the data.
+    shape : tuple
+        The shape of the data. A tuple of int (n_dipoles, n_times).
+    """
+
+    @verbose
+    def __init__(self, data, vertices=None, tmin=None, tstep=None, freqs=None,
+                 dims=("dipoles", "freqs", "times"), method=None, subject=None,
+                 verbose=None):
+
+        valid_dims = [("dipoles", "freqs", "times"),
+                      ("dipoles", "epochs", "freqs", "times"),
+                      ("dipoles", "orientations", "freqs", "times"),
+                      ("dipoles", "orientations", "epochs", "freqs", "times")]
+
+        valid_methods = ["morlet-power", "multitaper-power", "stockwell-power",
+                         "morlet-itc", "multitaper-itc", "stockwell-itc", None]
+
+        # unfortunately, _check option does not work with the original tuples
+        _check_option("dims", list(dims),
+                      [list(v_dims) for v_dims in valid_dims])
+        _check_option("method", method, valid_methods)
+        _validate_type(vertices, (np.ndarray, list), "vertices")
+
+        data, kernel, sens_data, vertices = self._prepare_data(data, vertices,
```

Comment:
It's `sens_data` because it's simply derived from SourceEstimate `sens_data`. Maybe it's a good idea to call it `sensor_data`, but then I would try to name it uniformly for both classes.
I added Parameters to `prepare_data()` now to explain both arrays better. But if you imply that we should use one array to store both of them, I'd say that this could lead to confusion really quickly for devs and users (bc of not knowing which representation is used at a given moment).

Reply:
sorry, I thought you introduced the `sens_data`. Then the answer should have been that you were doing it for historical reasons. For consistency with `_BaseSourceEstimate`. My fault.

Then I'm not sure how this should be tackle.